paasta_tools/autoscale_all_services.py

```diff
@@ -33,5 +34,5 @@ def main():
     autoscale_services(soa_dir)
 
 
-if __name__ == '__main__':
+if __name__ == "__main__" and is_mesos_leader():
```

Comment:
I've not seen this style used before. Any reason for doing this instead of adding the check inside of the various `main()` methods?